server/controllers/kaban.js

```diff
@@ -0,0 +1,287 @@
+const { createBoardColumn } = require("../utils/createBoardColumn");
+const { createColumnTask } = require("../utils/createColumnTask");
+const { deleteBoardBoard } = require("../utils/deleteBoardBoard");
+const { deleteBoardColumn } = require("../utils/deleteBoardColumn");
+const { deleteColumnTask } = require("../utils/deleteColumnTask");
+const { generateBoard } = require("../utils/generateBoards");
+const { updateBoardBoard } = require("../utils/updateBoardBoard");
+const { updateBoardColumn } = require("../utils/updateBoardColumn");
+const { updateColumnTask } = require("../utils/updateColumnTask");
+
+// @route GET /kaban/homeBoard
+// @access Protected
+exports.homeBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  let boards = await generateBoard(userId);
+  res.status(200)
+  res.json({ success: true, boards })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /kaban/board
+// @access Protected
+exports.createBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name } = req.body;
+  let board = await createBoardBoard({ name, userId });
+  res.status(200)
+  res.json({ success: true, board })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /kaban/board
+// @access Protected
+exports.updateBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name } = req.body;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let updatedBoard = await updateBoardBoard({ name }, board_id)
+  res.status(200)
+  res.json({ success: true, board: updatedBoard })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route DELETE /kaban/board
+// @access Protected
+exports.deleteBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  await deleteBoardBoard(board_id);
+  res.status(200)
+  res.json({ success: true })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /column/:board_id
+// @access Protected
+exports.createColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, index } = req.body;
```

Comment:
i believe we were able to get around this

// only allows 0 or null(i.e end) position within the Board
index = typeof index === 'number' ? 0 : null

I will assume it's corrected

Reply:
The idea of my comment is if there is no index we will append onto the list at index `n - 1` n being the length of the list. If there is an index it will only ever be at the beginning. 

Another approach is assuming 0 is left and 1 is right.

Reply:
0 = start 
1 = end

Reply:
// only allows 0 or 1(i.e end) position within the Column
    if (index == 1 || index == null) {
      index = null // mongodb takes null values as a directive to append to the end of the List
    } else {
      return res.status(403).json({
        succes: false,
        message: 'only end index is allowed during Card creation',
      })
    }

Reply:
I think we're overcomplicating this here. We can just do if index == 1 move onto the end, else shift onto the start. We shouldn't need to do an `else-if`